posix.mak

```diff
@@ -380,54 +385,69 @@ $(DOC_OUTPUT_DIR)/dmd-%.html : %.ddoc dcompiler.dd $(DDOC) $(DMD)
 $(DOC_OUTPUT_DIR)/dmd-%.verbatim : %.ddoc dcompiler.dd verbatim.ddoc $(DMD)
 	$(DMD) -c -o- -Df$@ verbatim.ddoc dcompiler.dd $<
 
+$(DOC_OUTPUT_DIR):
+	mkdir -p $@
+
 ################################################################################
 # Ebook
 ################################################################################
 
-dlangspec.d : $(SPEC_DD) ${STABLE_DMD}
+$G/dlangspec.d : $(SPEC_DD) ${STABLE_DMD}
 	$(STABLE_RDMD) ../tools/catdoc.d -o$@ $(SPEC_DD)
 
-dlangspec.html : $(DDOC) ebook.ddoc dlangspec.d $(DMD)
-	$(DMD) -conf= $(DDOC) ebook.ddoc dlangspec.d
+$G/dlangspec.html : $(DDOC) ebook.ddoc $G/dlangspec.d $(DMD)
+	$(DMD) -conf= -Df$@ $(DDOC) ebook.ddoc $G/dlangspec.d
 
-dlangspec.zip : dlangspec.html ebook.css
+$G/dlangspec.zip : $G/dlangspec.html ebook.css
 	rm -f $@
-	zip $@ dlangspec.html ebook.css
+	zip --junk-paths $@ $G/dlangspec.html ebook.css
```

Comment:
Without `--junk-paths` the zip would have `.generated/dlangspec.html` and not just `dlangspec.html`